interface/src/app/tribe_client/genesets.js

```diff
@@ -0,0 +1,88 @@
+angular.module('adage.tribe_client.genesets', [
+  'adage.tribe_client.resource'
+])
+
+// Geneset Search Factory
+// Allows for retreiving for and interacting with search results for genesets.
+.factory('GenesetSearch', function($rootScope, Genesets) {
+  var genesets = [];
+  var query = {};
+  var totalResults = 0;
+
+  return {
+    getQuery: function() {
+      return query;
+    },
+    getGenesets: function() {
+      return genesets;
+    },
+    clear: function() {
+      query = {};
+      genesets = [];
+      $rootScope.$broadcast('genesets.update');
+    },
+    totalResults: function() {
```

Comment:
I agree that `totalResults` sounds a bit like you're going to get all (the *totality*) of the GeneSet results. In addition, "total" could sound like a verb here, as if it's a calculation, but this is just an accessor. Perhaps a minor rework would allow this to be called `getResultCount()` and the corresponding attribute could be called `resultCount`?